gooey/gui/controller.py

```diff
@@ -85,16 +114,50 @@ def read_stdout(self, process, callback):
       if not line:
         break
       wx.CallAfter(self.core_gui.PublishConsoleMsg, line)
+      progress = self.progress_from_line(line)
+      if progress is not None:
+        wx.CallAfter(self.core_gui.UpdateProgressBar, progress)
     wx.CallAfter(callback, process)
 
+  def progress_from_line(self, text):
+    progress_regex = self.build_spec['progress_regex']

```

Comment:
It's a little tough to tell what the method is actually doing due to all the error handling boilerplate. I think if you pulled out some of the functionality into methods, you could expose the code's intent a little more clearly. For instance, factoring out the evaluation and conversion bits into their own handlers could give something like this:

```python

  def progress_from_line(self, text):
    progress_regex = self.build_spec['progress_regex'] or ''
    match = re.search(progress_regex, text.strip())
    if not match.group():
      return None
    progress_expr = self.build_spec['progress_expr']
    value = self.evaluate(progress_expr, match) if progress_expr else match.group(1)
    return safe_int(value)
```